tests/test_training.py

```diff
@@ -88,3 +88,27 @@ def test_train(device: str):
     dataset_name = create_c4_dataset_xsmall(device)
     test_cfg = gpt_tiny_cfg(dataset_name, device)
     main(test_cfg)
+
+
+def test_train_gauntlet():
+    """Test training run with a small dataset."""
+    os.chdir('scripts')
+    dataset_name = create_c4_dataset_xsmall('cpu')
+    test_cfg = gpt_tiny_cfg(dataset_name, 'cpu')
+    test_cfg.icl_tasks = ListConfig([
+        DictConfig({
+            'label':
+                'lambada_openai',
+            'dataset_uri':
+                'eval/local_data/language_understanding/lambada_openai.jsonl',
+            'num_fewshot': [0],
+            'icl_task_type':
+                'language_modeling'
+        })
+    ])
+    test_cfg.icl_subset_num_batches = 1  # -1 to evaluate on all batches
+    test_cfg.model_gauntlet = 'eval/yamls/model_gauntlet.yaml'
+    test_cfg.icl_seq_len = 128
+    test_cfg.max_duration = '1ba'
+    test_cfg.eval_interval = '1ba'
+    main(test_cfg)
```

Comment:
tests need some sort of assert or check

Reply:
I think ideally we would check that the ModelGauntlet was constructed and its eval_after_all method was called but I am not if it's possible to Mock that if the object is constructed within Main. Fwiw, the other unit test in this method has no asserts and simply checks whether the code fails

Reply:
Yeah, these tests are really "does it crash". For the gauntlet, there should separately be unit tests testing the basic functionality.

Reply:
I think it is important to test the full workflow of running train and seeing if it logs properly. 

I am trying to unit test this by using console logger but I think the definition of console logger's fit_end and eval_end methods preclude the gauntlet callback from working properly. This is because the gauntlet callback runs at eval_after_all. As it currently works, the console logger runs at eval_end and at fit end, but anything logged between eval_end and fit_end (e.g. during eval_after_all) will just sit in purgatory and not get flushed.

Lmk how you think I should proceed.

Reply:
I guess `ConsoleLogger` needs to flush on `eval_after_all`? Can you make a JIRA for that and assign it to Evan please?

In the meantime...can you test using `InMemoryLogger`?

Reply:
I'm not sure that InMemoryLogger will work. How will I access the logger's data from inside the test?

Reply:
I'm assuming in the test you are talking about you want to test calling `main`. Assuming that is true, you can return the `Trainer` from main, and then use the returned object to get whatever other objects you need.

Reply:
ok I will modify main to return trainer...good call